aws/iam_group.go

```diff
@@ -0,0 +1,137 @@
+				if shouldIncludeIamGroup(iamGroup, excludeAfter, configObj) {
+					allIamGroups = append(allIamGroups, iamGroup.GroupName)
+				}
+			}
+			return !lastPage
+		},
+	)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+	return allIamGroups, nil
+}
+
+//nukeAllIamGroups - delete all IAM Roles.  Caller is responsible for pagination (no more than 100/request)
+func nukeAllIamGroups(session *session.Session, groupNames []*string) error {
+	region := aws.StringValue(session.Config.Region) //Since this is a global resource this can be any random region
+	svc := iam.New(session)
+
+	if len(groupNames) == 0 {
+		logging.Logger.Info("No IAM Groups to nuke")
+		return nil
+	}
+
+	//Probably not required since pagination is handled by the caller
+	if len(groupNames) > 100 {
+		logging.Logger.Errorf("Nuking too many IAM Groups at once (100): Halting to avoid rate limits")
+		return TooManyIamGroupErr{}
+	}
+
+	//No bulk delete exists, do it with goroutines
+	logging.Logger.Info("Deleting all IAM Groups")
+	wg := new(sync.WaitGroup)
+	wg.Add(len(groupNames))
+	errChans := make([]chan error, len(groupNames))
+	for i, groupName := range groupNames {
+		errChans[i] = make(chan error, 1)
+		go deleteIamGroupAsync(wg, errChans[i], svc, groupName)
+	}
+	wg.Wait()
+
+	//Collapse the errors down to one
+	var allErrs *multierror.Error
+	for _, errChan := range errChans {
+		if err := <-errChan; err != nil {
+			allErrs = multierror.Append(allErrs, err)
+			logging.Logger.Errorf("[Failed] %s", err)
+		}
+	}
+	finalErr := allErrs.ErrorOrNil()
+	if finalErr != nil {
+		return errors.WithStackTrace(finalErr)
+	}
+
+	//Print Successful deletions
+	for _, groupName := range groupNames {
+		logging.Logger.Infof("[OK] IAM Group %s was deleted in %s", aws.StringValue(groupName), region)
+	}
+	return nil
+}
+
+//deleteIamGroup - removes an IAM group from AWS, designed to run as a goroutine
+func deleteIamGroupAsync(wg *sync.WaitGroup, errChan chan error, svc *iam.IAM, groupName *string) {
+	defer wg.Done()
+	var multierr *multierror.Error
+
+	//Remove any users from the group
+	//TODO make this threaded
+	getGroupInput := &iam.GetGroupInput{
+		GroupName: groupName,
+	}
+	grp, err := svc.GetGroup(getGroupInput)
+	for _, user := range grp.Users {
+		unlinkUserInput := &iam.RemoveUserFromGroupInput{
+			UserName:  user.UserName,
+			GroupName: groupName,
+		}
+		_, err := svc.RemoveUserFromGroup(unlinkUserInput)
+		if err != nil {
+			multierr = multierror.Append(multierr, err)
+		}
+	}
+
+	_, err = svc.DeleteGroup(&iam.DeleteGroupInput{
+		GroupName: groupName,
+	})
+	if err != nil {
+		multierr = multierror.Append(multierr, err)
+	}
+	errChan <- multierr.ErrorOrNil()
+}
+
+//check if iam group should be included based on config rules (RegExp and Exclude After)
+func shouldIncludeIamGroup(iamGroup *iam.Group, excludeAfter time.Time, configObj config.Config) bool {
+	if iamGroup == nil {
+		return false
+	}
+
+	if excludeAfter.Before(*iamGroup.CreateDate) {
```

Comment:
Prefer the AWS SDK convenience methods for dereferencing a pointer, i.e, [`aws.TimeValue`](https://github.com/aws/aws-sdk-go/blob/v1.44.107/aws/convert_types.go#L838)